Library/Homebrew/rubocops/python_internal.rb

```diff
@@ -0,0 +1,38 @@
+# typed: true
+# frozen_string_literal: true
+
+require "rubocops/extend/formula"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      # This cop audits `python-internal` dependencies in formulae.
+      class PythonInternalCop < FormulaCop
+        # python-internal can only be used by this list of formulae
+        ALLOWED_PYTHON_INTERNAL_USAGE = %w[
+          libxcb
+          xcb-proto
+        ].freeze
```

Comment:
Do we want this here, or in `audit_exceptions` at homebrew/core? I don't actually know why we'd prefer one over the other...

Reply:
I don't know. I copied this from the `uses_from_macos` cop so I though that's how things are done.

Reply:
CC @Rylan12

Reply:
It depends on whether this rule is something that should be enforced on all taps or just in homebrew/core and linuxbrew/core.

In the case of `uses_from_macos`, the list was kept here because third-party taps shouldn't be able to have their own list for what can be passed to `uses_from_macos`.

Without knowing the details of this particular case, I think it's probably a tap-specific decision. Theoretically, there could be formulae in third-party taps that _would_ want to depend on `python-internal` for the same reasons that `libxcb` and `xcb-proto` do.

Given that, it probably does make sense to have this list live in the `style_exceptions` directory of homebrew/core. Let me know if you need guidance on how to do that. There's a helped method [`tap_style_exception?`](https://github.com/Homebrew/brew/blob/2c83ea7339df5ae4d63b0eeb810f638f6d2cc7ca/Library/Homebrew/rubocops/extend/formula.rb#L155-L177) that can be used to check whether the given formula is an exception or not.

Reply:
Yes, I agree. Enforcing this audit in third-party taps is probably not what we want.

Reply:
Cool, makes sense. I updated the PR accordingly. I was able to update the test too.